processor/src/operations/crypto_ops.rs

```diff
@@ -186,6 +188,129 @@ impl Process {
 
         Ok(())
     }
+
+    // STREAM CIPHER OPERATIONS
+    // --------------------------------------------------------------------------------------------
+
+    /// Encrypts data from source memory to destination memory using RPO sponge keystream.
+    ///
+    /// This operation performs AEAD encryption by:
+    /// 1. Loading 8 elements (2 words) from source memory at stack[12]
+    /// 2. Adding each element to the corresponding rate element (stack[0..7])
+    /// 3. Writing the resulting ciphertext to destination memory at stack[13]
+    /// 4. Updating stack[0..7] with the ciphertext (becomes new rate for next hperm)
+    /// 5. Preserving capacity (stack[8..11])
+    /// 6. Incrementing both source and destination pointers by 8
+    ///
+    /// Stack transition:
+    /// [rate(8), cap(4), src_ptr, dst_ptr, ...] -> [ciphertext(8), cap(4), src_ptr+8, dst_ptr+8,
+    /// ...]
+    pub(super) fn op_crypto_stream(
+        &mut self,
+        err_ctx: &impl ErrorContext,
+    ) -> Result<(), ExecutionError> {
+        use miden_core::WORD_SIZE;
+
+        const SRC_PTR_IDX: usize = 12;
+        const DST_PTR_IDX: usize = 13;
+
+        let ctx = self.system.ctx();
+        let clk = self.system.clk();
+
+        // Get source and destination pointers from stack
+        let src_ptr = self.stack.get(SRC_PTR_IDX);
+        let dst_ptr = self.stack.get(DST_PTR_IDX);
+
+        // Calculate addresses for the two words
+        let src_word1_addr = src_ptr;
+        let src_word2_addr = src_ptr + Felt::from(WORD_SIZE as u32);
+        let dst_word1_addr = dst_ptr;
+        let dst_word2_addr = dst_ptr + Felt::from(WORD_SIZE as u32);
+
+        if src_ptr == dst_ptr {
```

Comment:
Doesn't this get enforced by the memory component?

Reply:
yes, though I think it is helpful to have it here as well as the error would be more informative
Though happy to remove it if you think otherwise